### PR TITLE
TECH: Add tooltip to help describe fields

### PR DESCRIPTION
**Pull request by tmadar**
Add a `tooltip` by input fields that pulls from the `description` attribute to help explain ambiguous sounding fields

image

Help to address feedback about a certain field faster than digging through the documentations tab

image

### CODE REVIEW

**Comment by matthew-kerle** [comment]
Oh nice! I was just thinking about talking to Jason about something like this actually the other day.

File: package.json [hunk 1]
```diff
@@ -60,8 +60,10 @@
     "arrowParens": "always"
   },
   "dependencies": {
+    "@material-ui/core": "4.11.3",
```
Comment (bjyoungblood) [inline package.json:63]: As much as I hardcore material-ui (and I really, really do), [`rc-tooltip`](https://github.com/react-component/tooltip) is a lot lighter, and we already package it with other projects.

File: src/types.ts [hunk 2]
```diff
@@ -70,6 +70,7 @@ export type Param = {
   json?: boolean; // value should be stringified, deprecated -- use `properties` for Objects and `array` for lists
   properties?: Param[]; // if working with json, pass in array of properties
   description?: string;
+  tooltip?: boolean;
```
Comment (bjyoungblood) [inline src/types.ts:73]: Looking at the code, it seems like if `tooltip` is false, we don't render `description` at all. Is that accurate? If that's the case, then could we just render the tooltip if `description` is not false-y? If not, then what do you think of changing tooltip to a string and treating it separately from `description`?

**Review by bjyoungblood — CHANGES_REQUESTED** [review]

File: src/types.ts (hunk 2 above)
Comment (tmadar) [reply src/types.ts:73]: i agree. @jwdotjs walked me through that we want the attributes to follow similar to OpenApiSpex, so I can simplify things and just check if a description is provided for deciding to render.

File: package.json (hunk 1 above)
Comment (tmadar) [reply package.json:63]: nice! I'll take a look at `rc-tooltip`. i used material-ui as its what I'm used to using in other projects

Comment (tmadar) [reply package.json:63]: Been trying to update to use `rc-tooltip` over material-ui and having a bear of a time trying to implement it without introducing my own sort of state management. I hover / click - the text shows - then clicking or unhovering doesnt cause it to disappear. Same sort of behavior found in this person's code sandbox from SO https://stackoverflow.com/questions/63628698/react-rc-tooltips-tooltips-wont-hide-after-state-changes

**Comment by tmadar** [comment]
Replace material-ui with rc-tooltip per @bjyoungblood suggestion and brought in the `help` icon from the js-shared icons. The component styling was borrowed from CMW `assets/js/react/common/HoverText.tsx`. I think maybe the icon component needs some changing in here because the fill-color looks off

Screen Shot 2021-04-24 at 9 16 08 AM

File: package.json [hunk 3]
```diff
@@ -60,8 +60,9 @@
     "arrowParens": "always"
   },
   "dependencies": {
+    "rc-tooltip": "5.1.0",
```
Comment (bjyoungblood) [inline package.json:63]:
```suggestion
    "rc-tooltip": "^5.1.0",
```

Requiring an exact version prevents dependents from deduplicating if they don't also use exactly 5.1.0.

File: src/common/Tooltip.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+              lineHeight: 1.35,
+              color: "white",
+            }}
+          >
+            {param.description}
+          </p>
+          </div>
+    }
+    >
+        <HelpIcon color={"rgb(66, 153, 225"} />
```
Comment (bjyoungblood) [inline src/common/Tooltip.tsx:43]:
```suggestion
        <HelpIcon color="rgb(66, 153, 225)" />
```

This might fix the fill color?

**Review by bjyoungblood — COMMENTED** [review]
Looking good. I think you just need to run `yarn prettier --write ./src` and it'll get a ✅  from me!

> I think maybe the icon component needs some changing in here because the fill-color looks off.

I think that might just be [how the icon looks](https://smartrent-ui.com/icons). 😕

File: package.json (hunk 3 above)
Comment (tmadar) [reply package.json]: done

File: src/common/Tooltip.tsx (hunk 4 above)
Comment (tmadar) [reply src/common/Tooltip.tsx]: nope. pushed up the change to it though

**Comment by tmadar** [comment]
@bjyoungblood 
> Looking good. I think you just need to run `yarn prettier --write ./src/**` and it'll get a ✅ from me!

done. thanks for the review and suggestions!

**Review by bjyoungblood — APPROVED** [review]

**Pull request merged**